com.unity.shadergraph/Editor/Data/Graphs/GraphData.cs

```diff
@@ -1482,6 +1482,27 @@ public void RemoveGraphInput(ShaderInput input)
             ValidateGraph();
         }
 
+        public void MoveCategory(CategoryData category, int newIndex)
+        {
+            if (newIndex > m_CategoryData.Count || newIndex < 0)
+                throw new ArgumentException("New index is not within categories list.");
```

Comment:
Should this use AssertHelpers instead? Do we really want to softlock the user's graph in this case?